test/unit/search/search_test.cpp

```diff
@@ -259,6 +259,21 @@ TYPED_TEST(search_test, error_levenshtein)
     }
 }
 
+TYPED_TEST(search_test, error_indel_no_substitution)
+{
+    using hits_result_t = std::vector<typename TypeParam::size_type>;
+
+    {
```

Comment:
can you add a comment which case this tests now.
Also there where a lot of new branches added. Are all of these covered with these two test instances?

Reply:
:+1: 
My changes made it more unlikely that the search calls itself recursively. I noticed now that I am missing some return true, they can only be reached during best mapping. So I added some additional test which will hopefully cover them.